app/src/main/java/fr/neamar/kiss/DataHandler.java

```diff
@@ -366,6 +366,16 @@ public void removeShortcuts(String packageName) {
         }
     }
 
+    @NonNull
+    public Set<String> getExcludedFromHistory() {
+        Set<String> excluded = PreferenceManager.getDefaultSharedPreferences(context).getStringSet("excluded-apps-from-history", null);
+        if (excluded == null) {
+            excluded = new HashSet<>();
+            excluded.add(context.getPackageName());
```

Comment:
Although not a terrible problem, this means that KISS is excluded until you exclude your first item. After that, KISS can be included :)

Reply:
Oh yes. kiss is excluded from its own history